File: services/search-service/src/main/java/com/visor/school/search/model/SearchIndex.java
package com.visor.school.search.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.elasticsearch.annotations.Document;
import org.springframework.data.elasticsearch.annotations.Field;
import org.springframework.data.elasticsearch.annotations.FieldType;

import java.util.Map;
import java.util.UUID;

/**
 * Search index document for Elasticsearch
 */
@Data
@Builder(toBuilder = true)
@NoArgsConstructor
@AllArgsConstructor
@Document(indexName = "school_search")
public class SearchIndex {

    @Id
    private UUID id;

    @Field(type = FieldType.Keyword)
    private SearchType type;

    @Field(type = FieldType.Text, analyzer = "standard")
    private String title;

    @Field(type = FieldType.Text, analyzer = "standard")
    private String content;

    @Field(type = FieldType.Object)
    private Map<String, Object> metadata;
}

File: services/search-service/src/main/kotlin/com/visor/school/search/model/SearchIndex.kt
package com.visor.school.search.model

import org.springframework.data.annotation.Id
import org.springframework.data.elasticsearch.annotations.Document
import org.springframework.data.elasticsearch.annotations.Field
import org.springframework.data.elasticsearch.annotations.FieldType
import java.util.UUID

/**
 * Search index document for Elasticsearch
 */
@Document(indexName = "school_search")
data class SearchIndex(
    @Id
    val id: UUID,

    @Field(type = FieldType.Keyword)
    val type: SearchType,

    @Field(type = FieldType.Text, analyzer = "standard")
    val title: String,

    @Field(type = FieldType.Text, analyzer = "standard")
    val content: String,

    @Field(type = FieldType.Object)
    val metadata: Map<String, Any> = emptyMap()
)

